src/app/components/problems/problem-set/problem-set.component.html

```diff
@@ -94,12 +94,20 @@
                 <a class="action-icon btn-primary faEye mr-2" routerLink="/problem/{{question.id}}">
                     <fa-icon [icon]="faEye"></fa-icon>
                 </a>
-                <a class="action-icon btn-warning faPencilAlt mr-2" routerLink="/problem/{{question.id}}/edit">
+                <a class="action-icon btn btn-warning faPencilAlt mr-2" routerLink="/problem/{{question.id}}/edit">
                     <fa-icon [icon]="faPencilAlt"></fa-icon>
                 </a>
-                <a class="action-icon btn-danger faTrashAlt" [routerLink]="" (click)="open(deleteModal, question.id)">
-                    <fa-icon [icon]="faTrashAlt"></fa-icon>
-                </a>
+                <ng-container *ngIf="question.author_name === user?.username; else cannotDelete">
```

Comment:
We have to compare author and user by id.

Reply:
We don't have access to id in the UI.